pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -114,6 +133,82 @@ func (m *manager) UpdatePodFromAllocation(pod *v1.Pod) (*v1.Pod, bool) {
 	return updatePodFromAllocation(pod, allocs)
 }
 
+func GetPodResizeAllocation(pod *v1.Pod) (*PodResourceSummary, bool) {
+	resizeManager.mutex.RLock()
+	defer resizeManager.mutex.RUnlock()
+
+	resizeAllocation, exists := resizeManager.podResizeAllocation[pod.UID]
+	if !exists {
+		return nil, false
+	}
+	return resizeAllocation, true
+}
+
+func ClearPodResizeAllocation(pod *v1.Pod) {
+	resizeManager.mutex.Lock()
+	defer resizeManager.mutex.Unlock()
+	delete(resizeManager.podResizeAllocation, pod.UID)
+}
+
+// Compare reqOld and reqNew, if different, store reqNew resource to diffReq
+func compareAndSaveDiff(reqOld, reqNew v1.ResourceRequirements) v1.ResourceRequirements {
+	var diffReq v1.ResourceRequirements
+
+	// Compare Requests resource
+	diffReq.Requests = make(v1.ResourceList)
+	for key, reqNewVal := range reqNew.Requests {
+		if reqOldVal, exists := reqOld.Requests[key]; !exists || !reqOldVal.Equal(reqNewVal) {
+			diffReq.Requests[key] = reqNewVal
+		}
+	}
+
+	// Compare Limits resource
+	diffReq.Limits = make(v1.ResourceList)
+	for key, reqNewVal := range reqNew.Limits {
+		if reqOldVal, exists := reqOld.Limits[key]; !exists || !reqOldVal.Equal(reqNewVal) {
+			diffReq.Limits[key] = reqNewVal
+		}
+	}
+	return diffReq
+}
+
+// Check Requests and Limists are empty
+func isResizeResourceEmpty(req v1.ResourceRequirements) bool {
+	return len(req.Requests) == 0 && len(req.Limits) == 0
+}
```

Comment:
This function probably is not needed anymore. It's used only once and it would be much clear to use this `if` statement directly.